trunk/SUAVE/Input_Output/OpenVSP/write_vsp_propeller.py

```diff
@@ -0,0 +1,197 @@
+## @ingroup Input_Output-OpenVSP
+# write_vsp_propeller.py
+# 
+# Created: Feb 2021, M. Clarke
+# ----------------------------------------------------------------------
+#  Imports
+# ---------------------------------------------------------------------- 
+from SUAVE.Core import Units, Data
+from SUAVE.Methods.Aerodynamics.AVL.purge_files import purge_files 
+from SUAVE.Methods.Geometry.Two_Dimensional.Cross_Section.Airfoil.import_airfoil_geometry\
+     import import_airfoil_geometry
+import numpy as np
+import shutil 
+
+## @ingroup Input_Output-OpenVSP
+def write_vsp_propeller(vsp_bem_filename,propeller):
+    """   This functions write a .bem file for OpenVSP
+
+    Assumptions:
+        None
+        
+    Source:
+        None
+
+    Inputs:
+        OpenVSP .bem filename
+        SUAVE Propeller Data Structure 
+
+    Outputs:
+        OpenVSP .bem file 
+
+    Properties Used:
+        N/A
+    """    
+    
+    # unpack inputs 
+    # Open the vsp_bem file after purging if it already exists
+    purge_files([vsp_bem_filename]) 
+    vsp_bem = open(vsp_bem_filename,'w')
+
+    with open(vsp_bem_filename,'w') as vsp_bem:
+        make_header_text(vsp_bem, propeller)
+         
+        make_section_text(vsp_bem,propeller)
+        
+        make_airfoil_text(vsp_bem,propeller)  
+            
+    return
+
+## @ingroup Input_Output-OpenVSP
+def make_header_text(vsp_bem,prop):  
+    """This function writes the header of the OpenVSP .bem file
+
+    Assumptions:
+        None
+        
+    Source:
+        None
+
+    Inputs:
+        vsp_bem - OpenVSP .bem file 
+        prop    - SUAVE propeller data structure 
+        
+    Outputs:
+        NA                
+
+    Properties Used:
+        N/A
+    """      
+    header_base = \
+'''...{0}... 
+Num_Sections: {1}
+Num_Blade: {2}
+Diameter: {3}
+Beta 3/4 (deg): {4}
+Feather (deg): 0.00000000
+Pre_Cone (deg): 0.00000000
+Center: {5}, {6}, {7}
+Normal: {8}, {9}, {10}
+
+''' 
+    # Unpack inputs 
+    name     = prop.tag
+    N        = len(prop.radius_distribution)
+    B        = prop.number_of_blades
+    D        = prop.tip_radius*2
+    beta     = np.round(prop.twist_distribution/Units.degrees,5)
+    X        = prop.origin[0][0]
+    Y        = prop.origin[0][1]    
+    Z        = prop.origin[0][2]    
+    Xn       = np.round(np.cos(np.pi- prop.thrust_angle ),5)
+    Yn       = 0.0000
+    Zn       = np.round(np.sin(np.pi- prop.thrust_angle ),5)
+    
+    beta_3_4 = beta[round(N*0.75)] 
```

Comment:
You could do an interpolation here to get the actual beta at the 75% chord. This might not be critical, but could cause some small errors in the future.

Reply:
changed to interpolation